README.md

```diff
@@ -94,6 +94,17 @@ Array of [rehype plugins](https://mdxjs.com/advanced/plugins#using-remark-and-re
 | `process` | `function` | A hook function whose return value will be appended to the processed front matter. This function is given access to the source `.mdx` content as the first parameter. |
 | `phase`   | `string`   | Used to specify when to run the `process` function. Eligible values are `prebuild`, `loader`, `both`. Defaults to `both` if not specified.                            |
 
+### scan
+
+> `object` | optional
+
+Object of scan objects containing the following parameters
+
+| Property    | Type                                       | Description                                                                                                                                                                                                                                                                                                                                          |
+| ----------- | ------------------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `pattern`   | `RegEx`                                    | A RegEx to use for scanning `.mdx` content, enables Layout customization                                                                                                                                                                                                                                                                             |
+| `transform` | `function(match: Array[]): any` _optional_ | An optional callback function that transforms the result of the match operation. This function is passed an Array of any matching `.mdx` content that is returned by [`content.match(pattern)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#Return_value) operation utilizing the `pattern` RegEx. |
+
```

Comment:
Let's make sure to add info here about how to use the results in your layout! Also, what happens if there's no layout and this option is active?

Reply:
May I suggest we add a link to some sort of other `.md` file? Maybe `scan.md` linked from here? I'm not sure what process to follow for lengthier 'docs'-esk rundowns, but I'm unsure how to explain without a pretty lengthier walk-through.

I started to do that sort of thing at length in the PR description, and it is LARGE. 

cc: @kylemac

Reply:
I usually will keep this type of thing in the readme, but if you'd like to move it out to another file and link there, that is totally fine. Whatever you feel is best!

Reply:
On second thought, yeah it isn't too bad. Let me know of any concerns in describing the process if I should not include `<script>` use case. Maybe that'd be better suited for a different location when multiple use-cases are worth calling out. 

https://github.com/hashicorp/next-mdx-enhanced/blob/jm.content-to-layout-scan-option/README.md#scanning-mdx-content

Reply:
```suggestion

See ["Scanning MDX Content"](#scanning-mdx-content) for more details.
```